### ISSUE TITLE
How can i have edited text box width on blur [components/text.js]

### ISSUE THREAD

**Issue by harshan89** [nhn/tui.image-editor#273]
## Version
v3.7.0

## Development Environment
Windows 64 / Chrome

## Current Behavior
How can i have edited text box width onblur, existing code giving same result for both
Textbox full width 
Trans Width 

```js
_onBlur() {
        const ratio = this.getCanvasRatio();
        const editingObj = this._editingObj;
        const editingObjInfos = this._editingObjInfos;
        const textContent = this._textarea.value;
        let transWidth = (editingObj.width / ratio) - (editingObjInfos.width / ratio);
        let transHeight = (editingObj.height / ratio) - (editingObjInfos.height / ratio);
}
```

## Expected Behavior
I need to align text box to left or right based on initial width after entering the text, so i need initial textbox with and with after change the text.

**Comment by harshan89** [comment]
Anyway i did the alignment for my use and it's working fine, let me know if i am wrong or any best way to do this. [ I updated Text.js component ]

```js
/**
     * Blur event handler
     * @private
     */
    _onBlur() {
        const ratio = this.getCanvasRatio();
        const editingObj = this._editingObj;
        const editingObjInfos = this._editingObjInfos;
        const textContent = this._textarea.value;
        let transWidth = (editingObj.width / ratio) - (editingObjInfos.width / ratio);
        let transHeight = (editingObj.height / ratio) - (editingObjInfos.height / ratio);
        let text = '';

        if (ratio === 1) {
            transWidth /= 2;
            transHeight /= 2;
        }

        this._textarea.style.display = 'none';

        editingObj.set({
            left: editingObjInfos.left + transWidth,
            top: editingObjInfos.top + transHeight
        });

        if (this._user.type !== 'admin') {
            text = textContent.length ? textContent : this._previousTextState;
            if (parseInt(this._previousTextCount, 10) < parseInt(this._textarea.value.length, 10)) {
                text = text.substring(0, this._previousTextState.length);
            }
        } else {
            text = textContent;
        }

        this.getCanvas().add(editingObj);

        const params = {
            id: snippet.stamp(editingObj),
            type: editingObj.type,
            text
        };
        this.fire(events.TEXT_CHANGED, params);

        // set canvas alignments

        const obj = this.graphics.getObject(params.id);
        const diff = (editingObjInfos.width - obj.width) / 2;
        const newObj = obj;
        if (newObj.textAlign === 'right') {
            newObj.left += diff;
        } else if (newObj.textAlign === 'left') {
            newObj.left -= diff;
        }
        this.graphics.removeObjectById(params.id);
        this.getCanvas().add(newObj);
    }

```

**Comment by jinwoo-kim-nhn** [comment]
Excellent. That's it.

Another option is to use the `Itext` provided by the fabric.

If you look at the code, you can take care of the branching with `useItext`.

However, using IText can be problematic in mobile environments. That's a choice.

![스크린샷 2019-11-22 오후 12 05 47](https://user-images.githubusercontent.com/35218826/69394815-6f8f5380-0d20-11ea-80e1-e6d626dc4331.png)

**Comment by jinwoo-kim-nhn** [comment]
Solved in v3.10.0
